extensions/markup-3.2.md

```diff
@@ -0,0 +1,131 @@
+# IRC Message Markup
+
+Copyright (c) 2015 Edward Powell <me@embolalia.com>.
+
+Unlimited redistribution and modification of this document is allowed
+provided that the above copyright notice and this permission notice
+remains in tact.
+
+## What message markup attempts to solve
+
+Currently, styled text in IRC messages is achieved through the use of control
+codes. These codes are bytes within the range of unprintable ASCII characters,
+with color formatting followed by printable characters. Clients must know what
+these codes are and handle them, even if they don't support the functionality,
+in order to avoid showing garbage to the user. Few, if any, keyboards natively
+support the entry of these codes, essentially requiring key bindings for entry.
+Moreover, there is no formal specification of how these codes should work,
+instead leaving the definition of this behavior to what is supported by clients
+*de facto*.
+
+## Overview of markup elements
+
+Message markup is defined through the use of a number of common characters. For
+the purposes of the below, text is considered "wrapped" by a string if it is
+immediately preceded and immediately followed by the given string, without
+containing the string.
+
+* Text which is wrapped by one or more asterisks (`*`) or underscores
+  (`_`) SHOULD be rendered as emphasized. If a client is unable to support one
+  form of emphasis, another form MAY be supported.
+  * Where the text is wrapped by a single asterisk (`*`), it should be rendered as
+    italic.
+  * Where the text is wrapped by two asterisks (`**`), it should be rendered as
+    bold.
+  * Where the text is wrapped by a single underscore (`_`), it should be
+    rendered as underlined.
+* Text wrapped by square brackets (`[` and `]`) which is immediately followed
+  by text which is wrapped by parentheses (`(` and `)`) SHOULD be rendered as a
+  link.
+  * The text of the link MUST be the text within the square brackets.
+  * The target of the link MUST be the text within the parentheses.
+  * If the target of the link contains a colon (`:`) or period (`.`), it should
+    be treated as a URI.
+  * If the target of the link contains neither a colon nor a period, it should
+    be treated as either a nickname or a channel on the network from which the
+    message was sent.
```

Comment:
I considered this. The main reason I didn't is that it would require the `CHANTYPES` value from `ISUPPORT` to be available within the parsing code, which may be difficult in some implementations. It's certainly open to discussion, though.